internal/provider/resource_template.go

```diff
@@ -193,12 +193,6 @@ func publishTemplate(ctx context.Context, d *schema.ResourceData, client *sp.Cli
 		if err != nil {
 			return err
 		}
-
```

Comment:
Hoisted up this code into the calling functions (`resourceTemplateCreate` and `resourceTemplateUpdate`)

Reply:
Now that this code is removed I don't think it makes sense to pass in `publish` anymore and we could just conditionally call this if publish is true in `resourceTemplateUpdate`

Reply:
Updated to conditionally call `publishTemplate()`